test/common/tracing/http_tracer_impl_test.cc

```diff
@@ -278,6 +279,33 @@ TEST(HttpConnManFinalizerImpl, NullRequestHeaders) {
   finalizer.finalize(*span);
 }
 
+TEST(HttpConnManFinalizerImpl, UpstreamClusterTagSet) {
+  std::unique_ptr<NiceMock<MockSpan>> span(new NiceMock<MockSpan>());
+  NiceMock<Http::AccessLog::MockRequestInfo> request_info;
+  auto upstream_host = std::make_shared<NiceMock<Upstream::MockHostDescription>>();
+  NiceMock<Upstream::MockClusterInfo> cluster_info;
+  const std::string expected_cluster_name = "<my_upstream_cluster>";
+
+  EXPECT_CALL(request_info, bytesReceived()).WillOnce(Return(10));
+  EXPECT_CALL(request_info, bytesSent()).WillOnce(Return(11));
+  Optional<uint32_t> response_code;
+  EXPECT_CALL(request_info, responseCode()).WillRepeatedly(ReturnRef(response_code));
+  EXPECT_CALL(request_info, upstreamHost()).WillRepeatedly(Return(upstream_host));
```

Comment:
can you reuse https://github.com/lyft/envoy/blob/3b20b62e43e2216845d1e436026aff6a2c5bebb5/test/mocks/http/mocks.h#L62 and do not create `auto upstream_host = std::make_shared<NiceMock<Upstream::MockHostDescription>>();` at all?